### PR TITLE
Fixes a bug associated with axes_world_coords for APE14 branch

### PR DESCRIPTION
**Pull request by yashrsharma44**
This PR fixes a bug associated with `axes_world_coords` in `ape14_rewrite` branch. Currently when a WCS with a `lat/lon` param is queried, a result of incorrect shape of `SkyCoords` are observed. So for fixing the bug, this PR has been made.
##
Ping @Cadair!

### CODE REVIEW

File: ndcube/ndcube.py [hunk 1]
```diff
@@ -258,33 +235,32 @@ def world_to_pixel(self, *quantity_axis_list):
         result = [u.Quantity(world_to_pixel[index], unit=u.pix) for index in range(self.wcs.pixel_n_dim)]
         return result[::-1]
 
-    def axis_world_coords(self, *axes, edges=False):
+    def axis_world_coords(self, axes=(), edges=False, skycoord=True):
```
Comment (Cadair) [inline ndcube/ndcube.py]: why the change from `*axes`?

File: ndcube/ndcube.py [hunk 2]
```diff
@@ -327,6 +303,10 @@ def axis_world_coords(self, *axes, edges=False):
                           for axis in new_int_axes]
         n_dependent_axes = [len(da) for da in dependent_axes]
 
+        # Store the lat/lon when skycoord=False
+        TS = None
```
Comment (Cadair) [inline ndcube/ndcube.py]: what is `TS`? Can we use a more descriptive name?

File: ndcube/ndcube.py [hunk 3]
```diff
@@ -355,21 +335,45 @@ def axis_world_coords(self, *axes, edges=False):
                           for k in dependent_axes[i]], indexing="ij")
                     for k, axis in enumerate(dependent_axes[i]):
                         quantity_list[axis] = dependent_pixel_quantities[k]
+
                 # Perform wcs translation
                 dependent_axes_coords = self.pixel_to_world(*quantity_list)
+
+                # Wrap the dependent_axes_coords into a list if a single quantity
+                dependent_axes_coords = dependent_axes_coords if isinstance(dependent_axes_coords, list) else [dependent_axes_coords]
```
Comment (Cadair) [inline ndcube/ndcube.py]:
```suggestion
                dependent_axes_coords = list(dependent_axes_coords)
```

File: ndcube/ndcube.py [hunk 4]
```diff
@@ -355,21 +335,45 @@ def axis_world_coords(self, *axes, edges=False):
...
+                                axes_coords[j] = dependent_axes_coords[j]
+                            else:
+                                k = dependent_axis - j
+                                if(k==0):
+                                    TS = dependent_axes_coords[j].data.lon.deg
+                                elif(k==1):
+                                    TS = dependent_axes_coords[j].data.lat.deg
+
+                                index_of_skcoord = j+1
+                                axes_coords[j] = dependent_axes_coords[j].data.lat.deg if k==0 else dependent_axes_coords[j].data.lon.deg
```
Comment (Cadair) [inline ndcube/ndcube.py]: Do we want to return values here? Don't we normally return quantities from this method?

```suggestion
                                axes_coords[j] = dependent_axes_coords[j].data.lat if k==0 else dependent_axes_coords[j].data.lon
```

File: ndcube/ndcube.py (hunk 1 above)
Comment (yashrsharma44) [reply ndcube/ndcube.py]: I was getting an error of using keyword arguments after `*axes`.

File: ndcube/ndcube.py (hunk 2 above)
Comment (yashrsharma44) [reply ndcube/ndcube.py]: Umm, that is for `Tx/Ty` of `skycoord`. The name can be improved

File: ndcube/ndcube.py (hunk 4 above)
Comment (yashrsharma44) [reply ndcube/ndcube.py]: This method should be used internally, and many times I had to convert it into `deg` in plotting, so that's the point of it.

**Comment by DanRyanIrish** [comment]
File conflicts, what's the status of this PR @yashrsharma44 and @Cadair ?

**Comment by Cadair** [comment]
An alternative solution to this is in the 2.0 branch now.
